frontend/packages/dev-console/src/components/add/layout/Masonry.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import * as cx from 'classnames';
+import Measure, { BoundingRect } from 'react-measure';
+import './MasonryLayout.scss';
+
+type MasonryProps = {
+  columnsCount: number;
+  children: React.ReactElement[];
+};
+
+type Size = BoundingRect;
+
+export const Masonry: React.FC<MasonryProps> = ({ columnsCount, children }) => {
+  const [heights, setHeights] = React.useState<Record<string, Size>>({});
+  const columns = columnsCount || 2;
+  const setHeight = (key: string, size: Size) => {
+    setHeights((old) => ({ ...old, [key]: size }));
+  };
+  const groupedColumns = Array.from({ length: columns }, () => ({
+    height: 0,
+    items: [] as React.ReactElement[],
+  }));
+
+  let added = false;
+  let allRendered = true;
+  React.Children.forEach(children, (item, itemIndex) => {
+    // Fill first row directly
+    if (itemIndex < columns) {
+      groupedColumns[itemIndex].height += heights[item.key as string]?.height || 0;
```

Comment:
With the change above we could simplify this here as well:

```suggestion
      groupedColumns[itemIndex].height += heights[item.key as string] || 0;
```